x-pack/plugins/ml/common/types/annotations.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// The Annotation interface is based on annotation documents stored in the
+// `.ml-annotations-6` index, accessed via the `.ml-annotations` alias.
+
+// Annotation document mapping:
+// PUT .ml-annotations-6
+// {
+//   "mappings": {
+//     "annotation": {
+//       "properties": {
+//         "annotation": {
+//           "type": "text"
+//         },
+//         "create_time": {
+//           "type": "date"
+//         },
+//         "create_username": {
+//           "type": "keyword"
+//         },
+//         "timestamp": {
+//           "type": "date"
+//         },
+//         "end_timestamp": {
+//           "type": "date"
+//         },
+//         "job_id": {
+//           "type": "keyword"
+//         },
+//         "modified_time": {
+//           "type": "date"
+//         },
+//         "modified_username": {
+//           "type": "keyword"
+//         },
+//         "result_type": {
```

Comment:
`result_type` was required when the idea was to put annotations in the results index, but now we've got a dedicated index I don't think this field is necessary.  Every document in the index can be assumed to be an annotation.

Reply:
I left it in because there's this idea of having discussion threads with multiple comments by different users attached to an annotation so there might be a need for different types some point in the future. However, not sure if `result_type` would be the right wording for that, what do you think? Should I remove it for now and add this mapping once that comments feature is added or should we keep it to have the mapping ready for that, let me know what you think.

Reply:
It sounds fine to keep a type field.  Maybe it should simply be called `type` though, since it's not a result?

Reply:
Agreed! Changed in https://github.com/elastic/kibana/pull/26034/commits/a985d52f328a7de16a90396d1550491408272428#diff-c5c6ac3dbb0e7c91b6d127aa06121b2cR44